components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -134,6 +137,12 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
         throw new Error('Unsupported operation. Use initialize.')
     }
 
+    protected async guardAccess(resource: GuardedResource, op: ResourceAccessOp) {
+        if (!(await this.resourceAccessGuard.canAccess(resource, op))) {
+            throw new ResponseError(ErrorCodes.PERMISSION_DENIED, "operation not permitted");
```

Comment:
Not sure whether error message should be more informative? Before it indicated a resource and operation. Not sure whether it was useful though.